torch/csrc/api/include/torch/data/datasets/chunk.h

```diff
@@ -352,9 +367,17 @@ class ChunkDataset final
     free_workers();
     preload_threads_.clear();
 
-    chunk_reader_.reset();
+    if (!load_checkpoint_){
+      chunk_reader_.reset();
+      chunk_sampler_.reset(chunk_reader_.chunk_count());
+    }
+    else {
+      torch::load(chunk_sampler_, options_.resume_from_file_);
+      
+      // After the checkpoint is loaded, mark the boolean to false to prevent future loading.
+      load_checkpoint_ = false;
+    }
```

Comment:
Uh why do we need this deferred loading business? Because when we start iterating in a data loader it always calls `reset()` at the beginning of an epoch?

Reply:
>Because when we start iterating in a data loader it always calls reset() at the beginning of an epoch?

That's exactly why we need to load here -- 
if 'load' option is set, the chunk_sampler_ needs to be resumed from the file for the first epoch when reset() is called by data loader. This way, ChunkDataset read chunks starting from the last saved point instead from the beginning. This is the core load logic. After the first epoch, any following epoch should just reset chunk_sampler_ as normal.

Please let me know if you have any questions about this functionality.